### PR TITLE
Fix a crash on x86 systems without `RDTSCP` instruction

### LINKED ISSUE #4573: Implement fiber.top() for ARM
After implementing #2694 find a way to do the same on ARM architecture

### LINKED ISSUE #5869: fiber-top: Consider to detect rdtscp existance
**Bug description**

Repro:

```
fiber = require('fiber')
fiber.top_enable()
fiber.top()
```

There is crash in case when no `rdtscp` on host

```
Program terminated with signal SIGILL, Illegal instruction.
```

```
#0  __rdtscp (__A=0x2d54488 <main_cord+72>) at /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/ia32intrin.h:105
#1  cpu_stat_start (stat=0x2d54478 <main_cord+56>) at /tarantool/src/lib/core/fiber.c:86
#2  0x0000000000680fe8 in cpu_stat_reset (stat=<optimized out>) at /tarantool/src/lib/core/fiber.c:107
#3  fiber_top_enable () at /tarantool/src/lib/core/fiber.c:1317
#4  0x0000000000663ccb in lbox_fiber_top_enable (L=<optimized out>) at /tarantool/src/lua/fiber.c:386
#5  0x00000000006a6bb7 in lj_BC_FUNCC ()
#6  0x000000000064aaf7 in execute_lua_eval (L=0x435e23f0) at /tarantool/src/box/lua/call.c:375
#7  0x00000000006a6bb7 in lj_BC_FUNCC ()
#8  0x00000000006b7ca6 in lua_pcall ()
#9  0x0000000000669e03 in luaT_call (L=0x435e23f0, nargs=<optimized out>, nreturns=<optimized out>) at /tarantool/src/lua/utils.c:1036
#10 0x000000000064b973 in box_process_lua (ret=0x7fedc557feb0, ctx=0x7fedc557fd80, handler=0x64aa10 <execute_lua_eval>)
    at /tarantool/src/box/lua/call.c:539
#11 box_lua_eval (expr=<optimized out>, expr_len=<optimized out>, args=args@entry=0x7fedc557fdf0, ret=ret@entry=0x7fedc557feb0)
    at /tarantool/src/box/lua/call.c:565
#12 0x00000000006484b2 in box_process_eval (request=0x7fedbe4101c0, port=0x7fedc557feb0) at /tarantool/src/box/call.c:147
#13 0x000000000055b8dd in tx_process_call (m=0x7fedbe410140) at /tarantool/src/box/iproto.cc:1615
#14 0x0000000000684d34 in cmsg_deliver (msg=msg@entry=0x7fedbe410140) at /tarantool/src/lib/core/cbus.c:375
#15 0x00000000006858f8 in fiber_pool_f (ap=<error reading variable: value has been optimized out>) at /tarantool/src/lib/core/fiber_pool.c:64
#16 0x0000000000558fcc in fiber_cxx_invoke(fiber_func, typedef __va_list_tag __va_list_tag *) (f=<optimized out>, ap=<optimized out>)
    at /tarantool/src/lib/core/fiber.h:773
#17 0x000000000067fb30 in fiber_loop (data=<optimized out>) at /tarantool/src/lib/core/fiber.c:869
#18 0x0000000000d230af in coro_init () at /tarantool/third_party/coro/coro.c:110
```

Restricted host
```
cat /proc/cpuinfo
processor  : 0
vendor_id  : GenuineIntel
cpu family  : 6
model    : 85
model name  : Intel Xeon Processor (Skylake, IBRS)
stepping  : 4
microcode  : 0x1
cpu MHz    : 2095.078
cache size  : 4096 KB
physical id  : 0
siblings  : 1
core id    : 0
cpu cores  : 1
apicid    : 0
initial apicid  : 0
fpu    : yes
fpu_exception  : yes
cpuid level  : 13
wp    : yes
flags    : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx pdpe1gb lm constant_tsc rep_good nopl cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 pku ospke avx512_vnni md_clear
bugs    : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs taa itlb_multihit
bogomips  : 4190.15
clflush size  : 64
cache_alignment  : 64
address sizes  : 46 bits physical, 48 bits virtual
power management:
```

### PR DESCRIPTION
**Pull request by Gumix**
###     fiber: use clock_monotonic64 instead of __rdtscp

clock_gettime(CLOCK_MONOTONIC) is implemented via the RDTSCP instruction
on x86 an has the following advantages over the raw instruction:

* It checks for RDTSCP availability in CPUID.
  If RDTSCP is not supported, it switches to RDTSC.
* Linux guarantee that clock is monotonic, hence, the CPU miss
  detection is not needed.
* It works on ARM.

As for disadvantage, this function is about 2x slower compared to a
single RDTSCP instruction. Performance degradation measured by the
[fiber switch benchmark](https://github.com/tarantool/tarantool/issues/2694#issuecomment-546381304) is about 7% for num_fibers = 100/1000.

Closes #4573
Closes #5869

### CODE REVIEW

File: src/lua/fiber.c [hunk 1]
```diff
@@ -285,10 +285,6 @@ lbox_fiber_top(struct lua_State *L)
 			      " fiber.top_enable() first");
 	}
 	lua_newtable(L);
-	lua_pushliteral(L, "cpu_misses");
```
Comment (locker) [inline src/lua/fiber.c:288]: This counter is [documented](https://www.tarantool.io/en/doc/latest/reference/reference_lua/fiber/#lua-function.fiber.top). You should create a doc bot request to update the documentation.

**Review by locker — COMMENTED** [review]
Looks good to me. See a couple of minor comments.

File: src/lua/fiber.c (hunk 1 above)
Comment (Gumix) [reply src/lua/fiber.c:288]: Yes, thats why this pull request is a draft so far. If we agreed on this, I’ll update doc etc.

**Review by sergepetrenko — APPROVED** [review]
Thanks! The patch looks good to me.
Speaking of `cpu_misses`, I think it's fine to delete this field. With a corresponding docbot request. Please, don't forget that it can't be deleted in 2.10. Only in upcoming 2.11

File: src/lib/core/clock.c [hunk 2]
```diff
@@ -35,23 +35,31 @@ double
 clock_realtime(void)
 {
 	struct timespec ts;
-	clock_gettime(CLOCK_REALTIME, &ts);
+	int rc = clock_gettime(CLOCK_REALTIME, &ts);
+	if (rc != 0)
+		return 0.0;
```
[the remark this answers is not in the document]
Comment (Gumix) [reply src/lib/core/clock.c]: Yes, here `clock_gettime` may fail only on:
- EFAULT tp points outside the accessible address space.
- EINVAL The clockid specified is invalid.

Changed to `panic_syserror()`.

File: src/lua/fiber.c (hunk 1 above)
Comment (Gumix) [reply src/lua/fiber.c:288]: Done.

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Cherry-picked the first three commits (the ones that fix bug #5869) to 2.10.

**Pull request merged**